internal/lookoutv2/repository/getjobs.go

```diff
@@ -0,0 +1,376 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/G-Research/armada/internal/common/database"
+	"github.com/G-Research/armada/internal/common/database/lookout"
+	"github.com/G-Research/armada/internal/lookoutv2/model"
+)
+
+type GetJobsRepository interface {
+	GetJobs(ctx context.Context, filters []*model.Filter, order *model.Order, skip int, take int) *GetJobsResult
+}
+
+type SqlGetJobsRepository struct {
+	db           *pgxpool.Pool
+	queryBuilder *QueryBuilder
+}
+
+type GetJobsResult struct {
+	Jobs  []*model.Job
+	Count int
+}
+
+type jobRow struct {
+	jobId              string
+	queue              string
+	owner              string
+	jobSet             string
+	cpu                int64
+	memory             int64
+	ephemeralStorage   int64
+	gpu                int64
+	priority           int64
+	submitted          time.Time
+	cancelled          sql.NullTime
+	state              int
+	lastTransitionTime time.Time
+	duplicate          bool
+	priorityClass      sql.NullString
+	latestRunId        sql.NullString
+}
+
+type runRow struct {
+	jobId       string
+	runId       string
+	cluster     string
+	node        sql.NullString
+	pending     time.Time
+	started     sql.NullTime
+	finished    sql.NullTime
+	jobRunState int
+	exitCode    sql.NullInt32
+}
+
+type annotationRow struct {
+	jobId           string
+	annotationKey   string
+	annotationValue string
+}
+
+func NewSqlGetJobsRepository(db *pgxpool.Pool) *SqlGetJobsRepository {
+	return &SqlGetJobsRepository{
+		db:           db,
+		queryBuilder: &QueryBuilder{lookoutTables: NewTables()},
+	}
+}
+
+func (r *SqlGetJobsRepository) GetJobs(ctx context.Context, filters []*model.Filter, order *model.Order, skip int, take int) (*GetJobsResult, error) {
+	var err error
+	tx, err := r.db.BeginTx(ctx, pgx.TxOptions{
+		IsoLevel:       pgx.RepeatableRead,
+		AccessMode:     pgx.ReadWrite,
+		DeferrableMode: pgx.Deferrable,
+	})
+	if err != nil {
+		log.WithError(err).Error("failed to start transaction")
+		return nil, err
+	}
+	defer func() {
```

Comment:
`r.db.BeginTxFunc` might make this a bit easier as you won't have to sort out handling the transaction cleanup in a defer.